modules/dhcp_native_ms/dhcp_native_ms_main.rb

```diff
@@ -74,6 +74,7 @@ def unused_ip(subnet_address, mac_address, from_address, to_address)
       client = dhcpsapi.get_client_by_mac_address(subnet_address, mac_address) rescue nil
       return client[:client_ip_address] unless client.nil?
 
+      raise NoMethodError.new("DhcpsApi::Server#get_free_ip_address is not available on Windows Server 2008 and earlier versions.") unless dhcpsapi.api_level == DhcpsApi::Server::DHCPS_WIN2012_API
```

Comment:
> NoMethodError's not appropriate here, could you use Proxy::DHCP::Error?

I'd like to be more specific with this error, perhaps even return a 501 instead of 400 for it? I can add something like  Proxy::DHCP::NotImplemented and use it here...